config_parser.cc

```diff
@@ -198,15 +202,23 @@ bool NginxConfigParser::Parse(std::istream* config_file, NginxConfig* config) {
       config_stack.top()->statements_.back().get()->child_block_.reset(
           new_config);
       config_stack.push(new_config);
+      
+      num_unmatched_open_braces++;
+
     } else if (token_type == TOKEN_TYPE_END_BLOCK) {
-      if (last_token_type != TOKEN_TYPE_STATEMENT_END) {
+      if ((last_token_type != TOKEN_TYPE_STATEMENT_END && last_token_type != TOKEN_TYPE_END_BLOCK) 
+        || num_unmatched_open_braces <= 0) {
         // Error.
+        printf("error found\n");
```

Comment:
Fixed in the new commit. Accidentally left that from when I was doing some unit testing.